### PR TITLE
migrate agent code from gx repo to gx cloud repo, add initial dockerfile

### PR DESCRIPTION
**Pull request by allensallinger**
migrate agent code from gx repo to gx cloud repo, add initial dockerfile

### CODE REVIEW

**Comment by allensallinger** [comment]
@cla-bot check

File: great_expectations_cloud/agent/actions/agent_action.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+from abc import abstractmethod
+from typing import Generic, Sequence, TypeVar
+
+from great_expectations.agent.models import CreatedResource, Event
```
Comment (Kilo59) [inline great_expectations_cloud/agent/actions/agent_action.py:4]: Need a find and replace on any `from great_expectations.agent` to `from great_expectations_cloud.agent`

File: great_expectations_cloud/agent/actions/data_assistants/utils.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    CreatedResource,
+    RunDataAssistantEvent,
+)
+from great_expectations.core.batch import BatchRequest
+from great_expectations.data_context.data_context import CloudDataContext
+from great_expectations.datasource.fluent import Datasource as FluentDatasource
+from great_expectations.exceptions import StoreBackendError
+from great_expectations.rule_based_profiler.data_assistant_result.data_assistant_result import (
+    DataAssistantResult,
+)
```
Comment (allensallinger) [inline great_expectations_cloud/agent/actions/data_assistants/utils.py:17]: @Kilo59 This is the snippet that was flagging a warning with the 88 character line limit.

Comment (Kilo59) [reply great_expectations_cloud/agent/actions/data_assistants/utils.py:17]: This one isn't actually being flagged.

https://github.com/great-expectations/cloud/actions/runs/5880832757/job/15947986228

**Review by Kilo59 — APPROVED** [review]

**Pull request merged**